packages/svelte/src/compiler/phases/3-transform/utils.js

```diff
@@ -279,6 +279,7 @@ export function clean_nodes(
 		trimmed.length === 1 &&
 		((first.type === 'RenderTag' && !first.metadata.dynamic) ||
 			(first.type === 'Component' &&
+				!state.options.hmr &&
```

Comment:
probably good to have a comment explaining a bit why we can't go without a wrapper in HMR mode

Reply:
Let's marge this for now so it unblocks people, we can always have a follow up PR to add comments.